osu.Game/Overlays/News/Sidebar/NewsSideBar.cs

```diff
@@ -0,0 +1,113 @@
+
+        [BackgroundDependencyLoader]
+        private void load(OverlayColourProvider colourProvider)
+        {
+            RelativeSizeAxes = Axes.Y;
+            Width = 250;
+            InternalChildren = new Drawable[]
+            {
+                new Box
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Colour = colourProvider.Background4
+                },
+                new Container
+                {
+                    RelativeSizeAxes = Axes.Both,
+                    Padding = new MarginPadding
+                    {
+                        Top = 20,
+                        Left = 50,
+                        Right = 30
+                    },
+                    Child = new FillFlowContainer
+                    {
+                        Anchor = Anchor.TopCentre,
+                        Origin = Anchor.TopCentre,
+                        Direction = FillDirection.Vertical,
+                        AutoSizeAxes = Axes.Both,
+                        Spacing = new Vector2(0, 20),
+                        Children = new Drawable[]
+                        {
+                            yearsPanel = new YearsPanel(),
+                            monthsFlow = new FillFlowContainer<MonthPanel>
+                            {
+                                AutoSizeAxes = Axes.Both,
+                                Direction = FillDirection.Vertical,
+                                Spacing = new Vector2(0, 10)
+                            }
+                        }
+                    }
+                }
+            };
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+
+            Metadata.BindValueChanged(metadata =>
+            {
+                monthsFlow.Clear();
+
+                if (metadata.NewValue == null)
+                {
+                    yearsPanel.Hide();
+                    return;
+                }
+
+                yearsPanel.Years = metadata.NewValue.Years;
+                yearsPanel.Show();
+
+                if (metadata.NewValue != null)
+                {
+                    var dict = new Dictionary<int, List<APINewsPost>>();
+
+                    foreach (var p in metadata.NewValue.NewsPosts)
+                    {
+                        var month = p.PublishedAt.Month;
+
+                        if (dict.ContainsKey(month))
+                            dict[month].Add(p);
+                        else
+                        {
+                            dict.Add(month, new List<APINewsPost>(new[] { p }));
+                        }
+                    }
+
+                    bool isFirst = true;
+
+                    foreach (var keyValuePair in dict)
+                    {
+                        monthsFlow.Add(new MonthPanel(keyValuePair.Value)
+                        {
+                            IsOpen = { Value = isFirst }
+                        });
```

Comment:
how does this ensure that the dictionary does not mess up month order by swapping them? i think it just doesn't? dictionaries have no inherent order, and expecting enumeration order to be the same as insertion order is in fact invoking undefined behaviour which can be broken at any time if dotnet upstream decides to do so. [see docs](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.dictionary-2?redirectedfrom=MSDN&view=net-5.0#remarks):

> For purposes of enumeration, each item in the dictionary is treated as a `KeyValuePair<TKey,TValue>` structure representing a value and its key. The order in which the items are returned is undefined.

Reply:
I guess I should perform sorting by key then to ensure it's in order?